azclishell/app.py

```diff
@@ -460,11 +459,25 @@ def handle_jmespath_query(self, text, continue_flag):
 
     def handle_scoping_input(self, continue_flag, cmd, text):
         if SELECT_SYMBOL['default'] in text:
-            default = text.partition(SELECT_SYMBOL['default'])[2].split()
-            value = self.handle_scoping(default)
-            print("defaulting: " + value)
-            cmd = cmd.replace(SELECT_SYMBOL['default'], '')
-            telemetry.track_ssg('default command', value)
+            default = text.partition(SELECT_SYMBOL['default'])[2]
+            if not text:
+                value = ''
+            else:
+                value = default
+
+            if self.default_command:
+                tree_val = self.default_command + " " + value
+            else:
+                tree_val = value
+
+            if in_tree(self.completer.command_tree, tree_val):
+                self.handle_scoping(value)
+                print("defaulting: " + value)
+                cmd = cmd.replace(SELECT_SYMBOL['default'], '')
+                telemetry.track_ssg('default command', value)
+            else:
+                print("not a valid value")
```

Comment:
Is this for if the scope specified is not valid?
If so, the error message should say so.